database/rrdcalc.c

```diff
@@ -603,6 +605,83 @@ void rrdcalc_unlink_and_free(RRDHOST *host, RRDCALC *rc) {
     rrdcalc_free(rc);
 }
 
+static void rrdcalc_labels_unlink_alarm_loop(RRDHOST *host, RRDCALC *alarms) {
+    RRDCALC *rc, *clean = NULL;
+    for (rc = alarms; rc; rc = rc->next) {
+        if (!rc->labels) {
+            continue;
+        }
+
+        if (clean) {
+            info("Health configuration for alarm '%s' cannot be applied, because the host %s does not have the label(s) '%s'",
+                  clean->name,
+                  host->hostname,
+                  clean->labels);
```

Comment:
Please add after this line the following missing call:

```
 rrdcalc_unlink_and_free(host, clean);
```

Reply:
@amoss I found more one problem, please, overwrite this rrdcalc.c with the file I sent to you with fixes.

Reply:
Done.